/*
 * Copyright 2022 Google LLC.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

package yggdrasil_decision_forests.learner.gradient_boosted_trees.proto;

// Representations of the internal memory of the early stopping utility class.
// This proto is used to serialize the status of the EarlyStopping class when
// training is interrupted and resumed.
message EarlyStoppingSnapshot {
  // The names of the fields match the names of the fields in EarlyStopping.
  optional float best_loss = 1;
  optional float last_loss = 2;
  repeated float last_metrics = 3 [packed = true];
  repeated float best_metrics = 4 [packed = true];
  optional int32 best_num_trees = 5;
  optional int32 last_num_trees = 6;
  optional int32 num_trees_look_ahead = 7;
  optional int32 trees_per_iterations = 8;
  optional int32 initial_iteration = 9;
}